circuits/src/zk_circuits/valid_match_mpc.rs

```diff
@@ -415,10 +437,18 @@ pub struct ValidMatchMpcWitness {
     pub order1: LinkableOrder,
     /// The first party's balance
     pub balance1: LinkableBalance,
+    /// The price that the first party agreed to execute at for their asset
+    pub price1: FixedPoint,
+    /// The maximum amount that the first party may match
+    pub amount1: Scalar,
```

Comment:
sorry, I'm a bit confused by the difference between `amount1`/`amount2` and `order1.amount`/`order2.amount`.
the only places where I see `amount1`/`amount2` be used is in the `max_minus_min` checks, which previously used `order1.amount`/`order2.amount`.
but it seems the `amount` field on the `order` struct has not been completely moved out to the `ValidMatchMpcWitness` as I see it still getting used in `validate_volume_constraints`

Reply:
The `amount` here is referring to a maximum possible match amount with the given, agreed upon price that is constrained either by the order volume or the balance. This change allows orders to be matched with undercapitalized balances. In testing I caught an issue with using `order.amount` still when constraining `max_minus_min_amount` so I'll change that.